ui/src/tool-panel/km-post/km-post-infobox.tsx

```diff
@@ -42,28 +43,28 @@ const KmPostInfobox: React.FC<KmPostInfoboxProps> = ({
 }: KmPostInfoboxProps) => {
     const { t } = useTranslation();
     const [showEditDialog, setShowEditDialog] = React.useState(false);
-    const [confirmingDraftDelete, setConfirmingDraftDelete] = React.useState<boolean>();
+    const [confirmingDraftDelete, setConfirmingDraftDelete] = React.useState(false);
     const updatedKmPost = useLoader(
         () => getKmPost(kmPost.id, publishType),
-        [kmPost, kmPostChangeTime, publishType],
+        [kmPost.id, kmPostChangeTime, publishType],
     );
     const [kmPostLength, setKmPostLength] = React.useState<number>();
+    const [kmPostLengthLoading, setKmPostLengthLoading] = React.useState(false);
     const changeTimes = useKmPostChangeTimes(kmPost.id);
 
     React.useEffect(() => {
-        setKmPostLength(undefined);
-        getKmLengths(publishType, kmPost.trackNumberId).then((details) => {
-            details.find((value) => value.kmNumber === kmPost.kmNumber)
-                ? setKmPostLength(
-                      details.filter((value) => value.kmNumber === kmPost.kmNumber)[0].length,
-                  )
-                : '';
-        });
-    }, [kmPost]);
+        setKmPostLengthLoading(true);
 
-    function isOfficial(): boolean {
-        return publishType === 'OFFICIAL';
-    }
+        getKmLengths(publishType, kmPost.trackNumberId)
+            .then((allKmLengths) => {
+                const kmLength = allKmLengths.find((value) => value.kmNumber === kmPost.kmNumber)
+                    ?.length;
+
+                setKmPostLength(kmLength);
+            })
+            .catch(() => setKmPostLength(undefined))
+            .finally(() => setKmPostLengthLoading(false));
+    }, [kmPost.trackNumberId, publishType]);
```

Comment:
Ei tämän tiketin asiaa suoraan, mutta voisikohan tämän hoitaa jotenkin `useLoaderWithStatus`illa tämän nykyisen `useEffect`in sijasta. Promiseillahan sekin pelaa, joten luulisi?